### PR TITLE
Add webpack build system and implement UI in Elm

### PR DESCRIPTION
**Pull request by brendanzab**

### CODE REVIEW

**Comment by brendanzab** [comment]
Still need to add back the styling!

File: src/client/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+require('./index.html');
+require('./styles.scss');
+
+var Elm = require('./Main.elm');
+var mountNode = document.getElementById('main');
+
+// The third value on embed are the initial values for incomming ports into Elm
+var app = Elm.Main.embed(mountNode);
```
Comment (Marwes) [inline src/client/index.js]: is this duplicated from src/client/app.js ?

**Comment by Marwes** [comment]
Cool! Had a bit of trouble getting this to run locally so it would be nice to maybe make building this more automatic as well letting everything either be installed and run through cargo or npm.

Got it to run after running these steps in any case.
```
npm install
npm install -g webpack
npm install -g elm
webpack
cargo run
```

**Comment by Marwes** [comment]
We should add some links back to this repo and the main repo as well.

**Comment by Marwes** [comment]
Nit: The result area could be a bit clearer, maybe add `Result` as a header to it and/or make or put it in a [panel](http://getbootstrap.com/components/#panels) or [well](http://getbootstrap.com/components/#wells) or something similar.

**Comment by brendanzab** [comment]
Just fyi - I'm using [Bootstrap 4](http://v4-alpha.getbootstrap.com) for this.

Here is the latest iteration:

screen shot 2016-10-02 at 3 26 57 am

For some reason there is no space between the selector and the eval button :/

Would like to use the Ace editor at some stage. There is an [elm package for it](https://github.com/DenisKolodin/elm-ace), but it hasn't been published to the repository yet.

**Comment by Marwes** [comment]
> Just fyi - I'm using Bootstrap 4 for this.

Ok, any particlar reason?

> For some reason there is no space between the selector and the eval button :/

Feels like that is often a problem with bootstrap (though then that is the only thing I have used)

> Would like to use the Ace editor at some stage.

Wanted to do this as well but apparently ace does not work with cdn, atleast not with some magic incantations that I don't know :/ . Didn't want to bother with setting it up locally either since I didn't really know what to use (I guess webpack is the right thing?)

**Comment by brendanzab** [comment]
You should be able to install it via npm.

Thought I'd use Bootstrap 4 seeing as it will be released sometime soon...

**Comment by Marwes** [comment]
Fair enough, only asked because I was curious, don't really know a lot about web stuff.

File: src/client/index.js (hunk 1 above)
Comment (brendanzab) [inline src/client/index.js]: woopsie - app.js should not have been committed!

**Comment by brendanzab** [comment]
It's not perfect, but it looks ok for now. Can't figure out the alignment on the 'Result' header....

**Comment by brendanzab** [comment]
Still find the Rust server rather crazy - might have to look into cleaning it up somewhat. ;)

**Comment by Marwes** [comment]
Looks good to me! Any more changes coming? Will update the server sometime later today with https://github.com/Marwes/gluon/pull/173 (as a git dependency). Could try and clean it up a bit as well.

**Comment by brendanzab** [comment]
No, probably good for now!

**Comment by brendanzab** [comment]
No, probably good for now!

**Pull request merged**